contrib/ledger-gen/src/programs.rs

```diff
@@ -0,0 +1,260 @@
+use {
+    solana_sdk::{
+        bpf_loader_upgradeable::{
+            self,
+            UpgradeableLoaderState,
+        },
+        system_instruction,
+        signature::{Keypair, Signer},
+        transaction::Transaction,
+        message::Message,
+        instruction::{Instruction, AccountMeta},
+        pubkey::Pubkey,
+    },
+    solana_client::{
+        connection_cache::ConnectionCache,
+        rpc_client::RpcClient,
+        send_and_confirm_transactions_in_parallel::{send_and_confirm_transactions_in_parallel_blocking, SendAndConfirmConfig}
+    },
+    solana_cli::{
+        program::calculate_max_chunk_size,
+    },
+    std::{
+        sync::Arc,
+    },
+};
+
+
+/// Sets Up Buffer Account in Order to Deploy Program
+///
+/// # Arguments
+/// * `client` - RPC Client
+/// * `payer` - Keypair of the Payer
+/// * `program_data` - Program Data
+///
+/// # Returns
+/// * `buffer_account` - Buffer Account Keypair
+///
+/// # Examples
+/// ```
+/// let buffer_account = programs::set_up_buffer_account_transactions(&client, &payer, &program_data);
+/// ```
+pub fn set_up_buffer_account_transactions(client: &Arc<RpcClient>, payer: &Keypair, program_data: &Vec<u8>) -> Keypair {
```

Comment:
Should probably make the side effects (creating and sending transactions relevant to account creation) clear here. Also rename this to `set_up_buffer_account`

Reply:
renaming